> @\\ofborg build mumble @\\ofborg build pothos
> 
> Appear broken by this, nixos x86_64.

Odd. Here's the `tail` of the build logs of `mumble` and `pathos`, it looks like the library headers aren't being added to the Store's closure.

<details><summary>Build logs</summary>
<p>

## `mumble`
```
CMake Error in src/mumble/CMakeLists.txt:
  Imported target "Poco::XML" includes non-existent path

    "/nix/store/7qrk9wmdfmm129p9y4fydr70rd59bmpk-poco-1.12.4/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in src/mumble/CMakeLists.txt:
  Imported target "Poco::XML" includes non-existent path

    "/nix/store/7qrk9wmdfmm129p9y4fydr70rd59bmpk-poco-1.12.4/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

## `pathos`

```
CMake Error in lib/CMakeLists.txt:
  Imported target "Poco::Foundation" includes non-existent path

    "/nix/store/7qrk9wmdfmm129p9y4fydr70rd59bmpk-poco-1.12.4/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.

```

</p>
</details>

<details><summary>Result of `ls /nix/store/7qrk9wmdfmm129p9y4fydr70rd59bmpk-poco-1.12.4`</summary>
<p>

```sh
$ ls /nix/store/7qrk9wmdfmm129p9y4fydr70rd59bmpk-poco-1.12.4
bin  lib
```

</p>
</details>

--- a/pkgs/applications/radio/pothos/default.nix
+++ b/pkgs/applications/radio/pothos/default.nix
@@ -1,6 +1,7 @@
 { lib
 , mkDerivation
 , fetchFromGitHub
+, fetchpatch
 , cmake
 , pkg-config
 , doxygen
@@ -33,6 +34,12 @@ mkDerivation rec {
   patches = [
     # spuce's CMakeLists.txt uses QT5_USE_Modules, which does not seem to work on Nix
     ./spuce.patch
+    # Poco had some breaking API changes in 1.12
+    (fetchpatch {
+      name = "poco-1.12-compat.patch";
+      url = "https://github.com/pothosware/PothosCore/commit/092d1209b0fd0aa8a1733706c994fa95e66fd017.patch";
+      hash = "sha256-bZXG8kD4+1LgDV8viZrJ/DMjg8UvW7b5keJQDXurfkA=";
+    })
   ];
 
   nativeBuildInputs = [ cmake pkg-config doxygen wrapQtAppsHook ];

--- a/pkgs/development/libraries/poco/default.nix
+++ b/pkgs/development/libraries/poco/default.nix
@@ -1,38 +1,21 @@
-{ lib, stdenv, fetchurl, fetchpatch, cmake, pkg-config, zlib, pcre, expat, sqlite, openssl, unixODBC, libmysqlclient }:
+{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, zlib, pcre2, expat, sqlite, openssl, unixODBC, libmysqlclient }:
 
 stdenv.mkDerivation rec {
   pname = "poco";
 
-  version = "1.11.1";
+  version = "1.12.4";
 
-  src = fetchurl {
-    url = "https://pocoproject.org/releases/${pname}-${version}/${pname}-${version}-all.tar.gz";
-    sha256 = "sha256-MczOYCAEcnAAO/tbDafirUMohMI9PNUJyG9HzzpeXSo=";
+  src = fetchFromGitHub {
+    owner = "pocoproject";
+    repo = "poco";
+    sha256 = "sha256-gQ97fkoTGI6yuMPjEsITfapH9FSQieR8rcrPR1nExxc=";
+    rev = "poco-${version}-release";
   };
-
-  patches = [
-    # Use GNUInstallDirs (https://github.com/pocoproject/poco/pull/3503)
-    (fetchpatch {
-      name = "use-gnuinstalldirs.patch";
-      url = "https://github.com/pocoproject/poco/commit/16a2a74f6c28c6e6baca2ba26b4964b51d8a1b74.patch";
-      sha256 = "sha256-mkemG8UemJEUQxae1trKakhnJFJW0AufDYFAbmnINbY=";
-      # Files not included in release tarball
-      excludes = [
-        "Encodings/Compiler/CMakeLists.txt"
-        "PocoDoc/CMakeLists.txt"
-        "NetSSL_Win/CMakeLists.txt"
-        "PDF/CMakeLists.txt"
-        "SevenZip/CMakeLists.txt"
-        "ApacheConnector/CMakeLists.txt"
-        "CppParser/CMakeLists.txt"
-      ];
-    })
-  ];
 
   nativeBuildInputs = [ cmake pkg-config ];
 
   buildInputs = [ unixODBC libmysqlclient ];
-  propagatedBuildInputs = [ zlib pcre expat sqlite openssl ];
+  propagatedBuildInputs = [ zlib pcre2 expat sqlite openssl ];
 
   outputs = [ "out" "dev" ];
 
@@ -42,6 +25,13 @@ stdenv.mkDerivation rec {
   cmakeFlags = [
     "-DPOCO_UNBUNDLED=ON"
   ];
+
+  postFixup = ''
+    grep -rlF INTERFACE_INCLUDE_DIRECTORIES "$dev/lib/cmake/Poco" | while read -r f; do
+      substituteInPlace "$f" \
+        --replace "$"'{_IMPORT_PREFIX}/include' ""
+    done
+  '';
 
   meta = with lib; {
     homepage = "https://pocoproject.org/";

--- a/pkgs/development/libraries/poco/default.nix
+++ b/pkgs/development/libraries/poco/default.nix
@@ -37,7 +37,7 @@ stdenv.mkDerivation rec {
     homepage = "https://pocoproject.org/";
     description = "Cross-platform C++ libraries with a network/internet focus";
     license = licenses.boost;
-    maintainers = with maintainers; [ orivej ];
+    maintainers = with maintainers; [ orivej tomodachi94 ];
     platforms = platforms.unix;
   };
 }

